workflow/controller/workflowpod.go

```diff
@@ -221,6 +222,39 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 		}
 	}
 
+	// Apply the patch string from template
+	if woc.wf.Spec.PodSpecPatch != "" || tmpl.PodSpecPatch != "" {
+
+		if tmpl.PodSpecPatch == "" {
+			tmpl.PodSpecPatch = woc.wf.Spec.PodSpecPatch
```

Comment:
This logic does not allow someone to have both a workflow pod spec patch as well as a template pod spec patch since the template level patch supersedes the workflow level. We need to support both at the same time, since the patches may be against different parts of the pod spec.